### PR TITLE
Change global search dropdown to Link

### PR DESCRIPTION
**Pull request by jloh02**
## Context
Currently, global search results cannot be opened in a new tab. This is due to the implementation using downshift's onChange rather than `<a>` tags or `<Link>`
#3466 

## Implementation
By switching from using the `onChange` listener from downshift to a <Link> tag, we can allow users to open search results in new tabs like a normal `<a>` link. Using `<Link>` mimics the `router.push` behavior where other UI components are not reloaded when the link is clicked normally.

Current context menu only shows back
Screenshot (67)

Proposed changes allows opening in new tab
Screenshot (66)

## Other Information
- First contribution and first time using React component
- Not sure if using the <Link> tag affected the CSS styles in any other way. Looks visually similar on my end

### CODE REVIEW

File: website/src/views/layout/GlobalSearch.tsx [hunk 1]
```diff
@@ -292,7 +277,6 @@ class GlobalSearch extends Component<Props, State> {
       <Downshift
         isOpen={isOpen}
         onOuterClick={this.onOuterClick}
-        onChange={this.onChange}
```
Comment (ZhangYiJiang) [inline website/src/views/layout/GlobalSearch.tsx:295]: This breaks keyboard navigation. One way to fix this is to keep the original logic and add `onClick={(evt) => evt.preventDefault()}` to the anchors so they still use the same logic when clicked (we also don't need them to be `Link` in that case but that's not a big deal)

Comment (jloh02) [reply website/src/views/layout/GlobalSearch.tsx]: Keyboard navigation fixed in c6d65db9430b0fd79d9c85eef455505e1f799020

**Review by zwliew — CHANGES_REQUESTED** [review]

File: website/src/views/layout/GlobalSearch.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-@import '~styles/utils/modules-entry';
+@import "~styles/utils/modules-entry";
```
[the remark this answers is not in the document]
Comment (jloh02) [reply website/src/views/layout/GlobalSearch.scss]: e7b177a827afcd601c3456498d166e1b9e2c0270

File: website/src/views/layout/GlobalSearch.tsx [hunk 3]
```diff
@@ -35,6 +26,25 @@ type State = {
...
+    case SEARCH_RESULT: {
+      // TODO: Move this into a proper function
+      const path = item.result === VENUE_RESULT ? '/venues' : '/courses';
+      return `${path}?q=${encodeURIComponent(item.term)}`;
+    }
+
+    default:
+      throw Error(`Unknown item type: ${item}`);
+  }
+};
```
[the remark this answers is not in the document]
Comment (jloh02) [reply website/src/views/layout/GlobalSearch.tsx]: Not sure why but a few weeks back the linter gave me warnings requiring me to put a `default`. Anyways resolved in 2acbd15ba237914a47876e948518b806c3921b7d

**Review by zwliew — APPROVED** [review]

File: website/src/views/layout/GlobalSearch.tsx [hunk 4]
```diff
@@ -35,6 +26,27 @@ type State = {
...
+
+    case SEARCH_RESULT: {
+      // TODO: Move this into a proper function
+      const path = item.result === VENUE_RESULT ? '/venues' : '/courses';
+      searchUrl = `${path}?q=${encodeURIComponent(item.term)}`;
+      break;
+    }
+  }
+  return searchUrl;
+};
```
Comment (zwliew) [inline website/src/views/layout/GlobalSearch.tsx:48]: It's cases like this when I think the `consistent-return` rule should be disabled, but eh -- this is fine. :)

**Comment by zwliew** [comment]
Thank you!

**Pull request merged**